sos/collector/transports/oc.py

```diff
@@ -0,0 +1,206 @@
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import json
+import tempfile
+import os
+
+from sos.collector.transports import RemoteTransport
+from sos.utilities import (is_executable, shell_out, sos_get_command_output,
+                           SoSTimeoutError)
+
+
+class OCTransport(RemoteTransport):
+    """This transport leverages the execution of commands via a locally
+    available and configured ``oc`` binary for OCPv4 environments.
+
+    OCPv4 clusters generally discourage the use of SSH, so this transport may
+    be used to remove our use of SSH in favor of the environment provided
+    method of connecting to nodes and executing commands via debug pods.
+
+    Note that this approach will generate multiple debug pods over the course
+    of our execution
+    """
+
+    name = 'oc'
+
+    @property
+    def connected(self):
+        up = sos_get_command_output(
+            "oc wait --timeout=0s --for=condition=ready pod/%s"
+            % self.pod_name
+        )
+        return up['status'] == 0
+
+    def get_node_pod_config(self):
+        """Based on our template for the debug container, add the node-specific
+        items so that we can deploy one of these on each node we're collecting
+        from
+        """
+        return {
+            "kind": "Pod",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "%s-sos-collector" % self.address.split('.')[0],
+                "namespace": "default"
+            },
+            "priorityClassName": "system-cluster-critical",
+            "spec": {
+                "volumes": [
+                    {
+                        "name": "host",
+                        "hostPath": {
+                            "path": "/",
+                            "type": "Directory"
+                        }
+                    },
+                    {
+                        "name": "run",
+                        "hostPath": {
+                            "path": "/run",
+                            "type": "Directory"
+                        }
+                    },
+                    {
+                        "name": "varlog",
+                        "hostPath": {
+                            "path": "/var/log",
+                            "type": "Directory"
+                        }
+                    },
+                    {
+                        "name": "machine-id",
+                        "hostPath": {
+                            "path": "/etc/machine-id",
+                            "type": "File"
+                        }
+                    }
+                ],
+                "containers": [
+                    {
+                        "name": "sos-collector-tmp",
+                        "image": "registry.redhat.io/rhel8/support-tools",
```

Comment:
Just a question from an OCP noob: is it OK to put distro-specific image name to distro-generic code / response?

Reply:
This transport is for OCP clusters only - which only ships on RHEL. This wouldn't be enabled for any other kubernetes-based platform, so I think we're good here.